### PR TITLE
add arg  to _jvm_import_external to align with bazel repo version

### PR DESCRIPTION
**Pull request by natansil**
bazel repo has [artifact_sha256](https://github.com/bazelbuild/bazel/blob/d8ffad56129e2c358c5b208edb62918c8ed1220c/tools/build_defs/repo/jvm.bzl#L56) arg:
This change aligns rules_scala version of `jvm_import_external` to also have this arg.
See https://github.com/bazelbuild/rules_scala/issues/737

### CODE REVIEW

File: scala/scala_maven_import_external.bzl [hunk 1]
```diff
@@ -208,7 +208,7 @@ jvm_import_external = repository_rule(
         "rule_name": attr.string(mandatory = True),
         "licenses": attr.string_list(mandatory = True, allow_empty = False),
         "jar_urls": attr.string_list(mandatory = True, allow_empty = False),
-        "jar_sha256": attr.string(),
+        "jar_sha256": attr.string(doc = "'jar_sha256' is deprecated. Please use 'artifact_sha256'"),
```
Comment (natansil) [inline scala/scala_maven_import_external.bzl]: @ittaiz is this what you meant by deprecation?

Comment (ittaiz) [reply scala/scala_maven_import_external.bzl]: I meant https://github.com/bazelbuild/bazel/issues/5654
TLDR:
```
set the default value of the attribute to None and then in the rule implementation check if the attribute is set to anything other than None and print a warning to the user
```

Comment (natansil) [reply scala/scala_maven_import_external.bzl]: done. string type can't get None
